src/views/AnnualReport.vue

```diff
@@ -930,76 +924,48 @@ export default {
         )
       }
 
-      if (this.filingId > 0) {
-        // we have a filing id, so we are updating an existing filing
-        let url = `businesses/${this.entityIncNo}/filings/${this.filingId}`
-        if (isDraft) {
-          url += '?draft=true'
+      try {
+        let response
+
+        if (this.filingId > 0) {
+          // we have a filing id, so update (put) an existing filing
+          let url = `businesses/${this.entityIncNo}/filings/${this.filingId}`
+          if (isDraft) { url += '?draft=true' }
+          response = await axios.put(url, data)
+        } else {
+          // filing id is 0, so create (post) a new filing
+          let url = `businesses/${this.entityIncNo}/filings`
+          if (isDraft) { url += '?draft=true' }
+          response = await axios.post(url, data)
         }
-        let filing = null
-        await axios.put(url, data).then(res => {
```

Comment:
Since the bodies of PUT and POST calls were the same here (only the URLs and axios calls were different), I merged the bodies to reduce code duplication.

You rest of the PUT and POST bodies is substantially similar to what you saw in the Create UI and Edit UI changes yesterday.